src/main_utils.hpp

```diff
@@ -0,0 +1,66 @@
+//===--------------------------- main_utils.hpp ---------------------------===//
```

Comment:
I wonder if we should conform to the new file naming scheme like `MainUtils.hpp`?
For main.cpp, it was sort of reasonable to keep it this way for obvious reasons, moreover, users don't need to import this file anyways. OTOH, users are expected to import this header file to construct their own drivers.